core/src/main/scala/org/apache/spark/scheduler/TaskInfo.scala

```diff
@@ -93,6 +104,8 @@ class TaskInfo(
 
   def running: Boolean = !finished
 
+  def needResubmit: Boolean = !killedAttempt
```

Comment:
this looks unnecessary, we can just write `!killedByAttempt`

Reply:
Since `killedAttempt`/`killedByAttempt` is never used except in this negated sense, it probably makes more sense to get rid of it and this `needResubmit` method entirely, and just have a `needResubmit` var with the opposite sense of `killedAttempt` -- but I'm also not all that happy with the `needResubmit` name.